### DISCUSSION TITLE
no data over MQTT

### DISCUSSION THREAD

**Discussion by shizen1** [Koenkk/zigbee2mqtt#10883] [General]
Hello,

I have added a Tuya TS0201.
In the frontend I also see the data of the sensor.

![2022-01-18 19 38 55](https://user-images.githubusercontent.com/97977152/149998587-d5b563f5-99f5-4b23-9a2d-2ae9b0fbe762.png)

But via MQTT no messages come except "online".

![2022-01-18 19 39 30](https://user-images.githubusercontent.com/97977152/149998638-6708746e-5938-41c6-8526-c0c9ec0deb66.png)

Does anyone have any idea?

greetings

**Comment by shizen1** [comment]
Waiting until the temperature changes helps :-o

**Reply by electric-m** [reply]
It actually doesn’t .
You can provoke that by exhaling a few times in front of the sensor.

I have a similar issue here.

Xiaomi Temp Humidity works just fine, and I receive the values over MQTT in Iobroker MQTT Client.

Any other MQTT Client though only receives the „availability“ topic.

**Reply by shizen1** [reply]
I was able to narrow down the problem with the http://mqtt-explorer.com.

**Reply by electric-m** [reply]
Really? I tried MQTT EXPLORER too, but that didn’t help. What did you find out?

**Reply by shizen1** [reply]
I was able to quickly see the MQTT messages and drill down. So I was able to isolate my problem by different action on the senor and then integrate it well in Openhab.
I have not understood your problem yet :-(

**Reply by electric-m** [reply]
I started a new discussion for my problem, since it might differ from yours alltogehter: #10891

**Reply by electric-m** [reply]
> I was able to quickly see the MQTT messages and drill down. So I was able to isolate my problem by different action on the senor and then integrate it well in Openhab.
> I have not understood your problem yet :-(

Would still be happy to here, what exactly you changed to get it working.

**Reply by shizen1** [reply]
I used MQTT Explorer to monitor the traffic.
Then challenged the sensor with temperature changes.
Then the message appeared in the MQTT Explorer.
The frontend to the zigbee2mqtt service had data from the sensor all along. Is it the same for you? The frontend helps with troubleshooting.

**Reply by electric-m** [reply]
Yeah, sensor works just fine in frontend. I am coming to think, that it has to do with retaining.

I set retain to true and 3600 seconds now for this sensor, and it seems good for now.

**Comment by mmaric55** [comment]
hello,
hope you could help me. i have sensor Tuya ZG-227ZL (TS0601). i see all data in the frontend.
my problem is that these values are transferred to the iobroker in one row instead separate topic/values.

![image](https://user-images.githubusercontent.com/119892636/205905833-c9f02b65-240a-4053-afe2-ae8e7c860b94.png)

![image](https://user-images.githubusercontent.com/119892636/205905941-a948963a-d2e1-43ba-89e2-12d2030a5b72.png)


What could i do ?
Really appreaciate your help.

**Comment by mmaric55** [comment]
hi, i have solved it by my own. Thanks.
